src/index.tsx

```diff
@@ -137,14 +137,7 @@ export default class InfiniteScroll extends Component<Props, State> {
     }
   }
 
-  UNSAFE_componentWillReceiveProps(props: Props) {
-    // do nothing when dataLength and key are unchanged
-    if (
-      this.props.key === props.key &&
-      this.props.dataLength === props.dataLength
-    )
-      return;
-
+  UNSAFE_componentWillReceiveProps() {
```

Comment:
but this will setState on every update

Reply:
this.props.dataLength === props.dataLength

if dataLength is not changed, it will return, hence setState will have no chance to fire